omroneipApp/src/drvOmroneip.cpp

```diff
@@ -64,78 +20,89 @@ static int optimiseTagsC(void *drvPvt)
   return status;
 }
 
-omronEIPPoller::omronEIPPoller(const char *portName, const char *pollerName, double updateRate):
-                              belongsTo_(portName),
-                              pollerName_(pollerName),
-                              updateRate_(updateRate)
+omronEIPPoller::omronEIPPoller(const char *portName, const char *pollerName, double updateRate, int spreadRequests) : belongsTo_(portName),
+                                                                                                                      pollerName_(pollerName),
+                                                                                                                      updateRate_(updateRate),
+                                                                                                                      spreadRequests_(spreadRequests),
+                                                                                                                      myTagCount_(0)
 {
 }
 
+/* Gives the read pollers time to finish their processing loop and sets omronExiting_ to true to tell the pollers to destruct */
 static void omronExitCallback(void *pPvt)
 {
   drvOmronEIP *pDriver = (drvOmronEIP *)pPvt;
   pDriver->omronExiting_ = true;
-  epicsThreadSleep(1);
+  epicsThreadSleep(3);
 }
 
 /* This function is called by the IOC load system after iocInit() or iocRun() have completed */
 static void myInitHookFunction(initHookState state)
 {
-  switch(state) {
-    case initHookAfterIocRunning:
-      iocStarted = true;
-      break;
-    default:
-      break;
+  switch (state)
+  {
+  case initHookAfterIocRunning:
+    iocStarted = true;
+    break;
+  default:
+    break;
   }
 }
 
 drvOmronEIP::drvOmronEIP(const char *portName,
-                          char *gateway,
-                          char *path,
-                          char *plcType,
-                          int debugLevel)
+                         char *gateway,
+                         char *path,
+                         char *plcType,
+                         int debugLevel,
```

Comment:
This is only used to enable libplctags debugging. I suppose I could somehow monitor the current asynTrace level and update the libplctag debug level from this? I thought it might make more sense to keep the two seperate, as really there should be no need to debug libplctag and also their debugging levels are quite different. Libplctag can be very spammy which would just block out my asyn FLOW comments. Connecting the error & warning levels makes more sense, but the user should not have to worry about libplctag warnings really.

Reply:
Aha, I see. That makes sense to have this as a separate variable so that you can control libplctag debug output.
I would not recommend to use the asyn log levels for this, actually. But this constructor also doesn't feel like its the right place either. Instead, I would suggest using an EPICS IOC shell variable (iocsh var command) to control the debug level for libplctag.

The documentation for this is a little bit difficult to find, but essentially:

The database definition keyword variable forces a reference to an integer or double variable, e.g. debugging variables.
The xxxInclude.dbd file can contain definitions like:
```
variable(asCaDebug,int)
variable(myDefaultTimeout,double)
```
The code that defines the variables must include code like:
```
int asCaDebug = 0;
epicsExportAddress(int,asCaDebug);
```

☝️ from section 2.5.1 https://epics.anl.gov/base/R3-14/12-docs/AppDevGuide.pdf 
See also section 6.11.